tools/decode.py

```diff
@@ -45,92 +53,51 @@ def run(file, **kwargs):
-                                c_pass = []
-
-                            if prev_goal_position == 0:
-                                # Second half
-                                if round(servo.goalPosition) > prev_goal_position:
-                                    # increasing
-                                    state = 'inc2stop'
-                                else:
-                                    # decreasing
-                                    state = 'dec2stop'
-                            elif round(servo.goalPosition) == 0:
-                                # First half
-                                if round(servo.goalPosition) > prev_goal_position:
-                                    # increasing
-                                    state = 'inc1stop'
-                                else:
-                                    # decreasing
-                                    state = 'dec1stop'
-                            elif round(servo.goalPosition) > prev_goal_position:
-                                # increasing
-                                state = 'inc'
-                            else:
-                                # decreasing
-                                state = 'dec'
-
-                            prev_goal_position = round(servo.goalPosition)
-
-                        c_pass.append({
-                            'error_flags': servo.errorFlags,
-                            'present_position': servo.presentPosition,
-                            'goal_position': servo.goalPosition,
-                            'present_speed': servo.presentSpeed,
-                            'goal_speed': servo.movingSpeed,
-                            'load': servo.load,
-                            'voltage': servo.voltage,
-                            'temperature': servo.temperature
-                        })
-
-                    else:
-                        out = re.sub(r'\s+', ' ', MessageToJson(msg, True))
-                        out = '{{ "type": "{}", "timestamp": {}, "data": {} }}'.format(parsers[type_hash][0].decode('utf-8'), timestamp, out)
-                        # Print as a json object
-                        print(out)
+
+        # While we can read a header
+        while len(f.read(3)) == 3:
+
+            # Read our size
+            size = struct.unpack('<I', f.read(4))[0]
+
+            # Read our payload
+            payload = f.read(size)
+
+            # Read our timestamp
+            timestamp = struct.unpack('<Q', payload[:8])[0]
+
+            # Read our hash
+            type_hash = payload[8:16]
+
+            # If we know how to parse this type, parse it
+            if type_hash in parsers:
+                msg = parsers[type_hash][1].FromString(payload[16:])
+
+                # So the else can always exist
+                if False:
+                    pass
+
+                # If there is a special way to decode a particular message do it here
```

Comment:
Can you explain a situation where this would be needed? Is it worth having this empty block here?